Wikipedia/Code/ExploreViewController.swift

```diff
@@ -927,7 +905,71 @@ class ExploreViewController: ColumnarCollectionViewController, ExploreCardViewCo
     }
 
     var addArticlesToReadingListVCDidDisappear: (() -> Void)? = nil
-    
+
+    // TODO: Remove after expiry date (1 March 2025)
+    private func presentYearInReviewAnnouncement() {
+
+        if UIDevice.current.userInterfaceIdiom == .pad && navigationBar.hiddenHeight > 0 {
+            return
+        }
+
+        let languages = ["fr", "it"]
+        guard YearInReviewFeatureAnnouncementTimeBox.isAnnouncementActive() else {
+            return
+        }
+
+        guard let appLanguage = dataStore.languageLinkController.appLanguage else {
+            return
+        }
+
+        guard languages.contains(appLanguage.languageCode) else {
+            return
+        }
+
+        guard let yirDataController = try? WMFYearInReviewDataController() else {
+            return
+        }
+
+        let project = WMFProject.wikipedia(WMFLanguage(languageCode: appLanguage.languageCode, languageVariantCode: nil))
+        guard yirDataController.shouldShowYearInReviewEntryPoint(countryCode: Locale.current.region?.identifier, primaryAppLanguageProject: project) else {
+            return
+        }
+
+        guard !yirDataController.hasPresentedYiRFeatureAnnouncementModel else {
```

Comment:
We might be able to go a bit further and create a `shouldShowYearInReviewFeatureAnnouncement` method in the data controller, which references this flag, the `YearInReviewFeatureAnnouncementTimeBox` value and the `shouldShowYearInReviewEntryPoint` response. Just to try reduce the amount of YiR code that has to live in ExploreViewController.